core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -28,19 +30,20 @@
 
     public ThreadPoolExecutor createPool(String name, ThreadingProfile tp)
     {
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
Not sure I'm following this change: did you removed the creation of an ImmutableThreadingProfile from all the other places and just added it here?